statistics/handle/update.go

```diff
@@ -706,18 +706,17 @@ func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) {
 		for _, tbl := range tbls {
 			tblInfo := tbl.Meta()
 			pi := tblInfo.GetPartitionInfo()
-			tblName := "`" + db + "`.`" + tblInfo.Name.O + "`"
 			if pi == nil {
 				statsTbl := h.GetTableStats(tblInfo)
-				sql := fmt.Sprintf("analyze table %s", tblName)
+				sql := "analyze table `" + db + "`.`" + tblInfo.Name.O + "`"
 				analyzed := h.autoAnalyzeTable(tblInfo, statsTbl, start, end, autoAnalyzeRatio, sql)
 				if analyzed {
 					return
 				}
 				continue
 			}
 			for _, def := range pi.Definitions {
-				sql := fmt.Sprintf("analyze table %s partition `%s`", tblName, def.Name.O)
+				sql := "analyze table `" + db + "`.`" + tblInfo.Name.O + "`" + " partition" + def.Name.O
```

Comment:
```suggestion
				sql := "analyze table `" + db + "`.`" + tblInfo.Name.O + "`" + " partition" + "`" + def.Name.O + "`"
```
partition name also need surrounding with "`"